lib/modules/versioning/ubuntu/index.ts

```diff
@@ -16,13 +16,25 @@ const di = new DistroInfo('data/ubuntu-distro-info.json');
 // validation
 
 function isValid(input: string): boolean {
-  return (
-    (typeof input === 'string' &&
-      regEx(/^(0[4-5]|[6-9]|[1-9][0-9])\.[0-9][0-9](\.[0-9]{1,2})?$/).test(
-        input
-      )) ||
-    di.isCodename(input)
-  );
+  if (typeof input !== 'string') {
+    return false;
+  }
+
+  if (
+    regEx(/^(0[4-5]|[6-9]|[1-9][0-9])\.[0-9][0-9](\.[0-9]{1,2})?$/).test(input)
+  ) {
+    return true;
+  }
+
+  if (di.isCodename(input)) {
+    return true;
+  }
+
+  return isDatedCodeName(input);
+}
+
+function isDatedCodeName(input: string): boolean {
```

Comment:
please move these new internal helper functopn to a `./common.ts` file, so this file is smaller and easier to maintain.

Reply:
have moved them. I left `getVersionByCodename` where it as as that interacts with DistroInfo and didn't think it'd be worth creating a new instance of that in the other file.